client/multi-currency/blocks/currency-switcher.js

```diff
@@ -0,0 +1,321 @@
+/**
+ * Internal dependencies
+ */
+import { useEnabledCurrencies, useCurrencies } from 'wcpay/data';
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { __ } from '@wordpress/i18n';
+import {
+	CheckboxControl,
+	PanelBody,
+	RangeControl,
+} from '@wordpress/components';
+import { registerBlockType } from '@wordpress/blocks';
+import {
+	ColorPaletteControl,
+	InspectorControls,
+	useBlockProps,
+} from '@wordpress/block-editor';
+
+registerBlockType( 'woocommerce-payments/multi-currency-switcher', {
+	apiVersion: 2,
+	title: __( 'Currency Switcher Block', 'woocommerce-payments' ),
+	description: __(
+		'Let your customers switch between your enabled currencies',
+		'woocommerce-payments'
+	),
+	icon: 'money-alt',
+	category: 'widgets',
+	attributes: {
+		symbol: {
+			type: 'boolean',
+			default: true,
+		},
+		flag: {
+			type: 'boolean',
+			default: false,
+		},
+		fontSize: {
+			type: 'integer',
+			default: 13,
+		},
+		fontLineHeight: {
+			type: 'float',
+			default: 1.5,
+		},
+		fontColor: {
+			type: 'string',
+			default: '#000000',
+		},
+		border: {
+			type: 'boolean',
+			default: true,
+		},
+		borderRadius: {
+			type: 'integer',
+			default: 3,
+		},
+		borderColor: {
+			type: 'string',
+			default: '#000000',
+		},
+		backgroundColor: {
+			type: 'string',
+			default: 'transparent',
+		},
+	},
+	edit: ( props ) => {
+		const {
+			attributes: {
+				symbol,
+				flag,
+				fontSize,
+				fontLineHeight,
+				fontColor,
+				border,
+				borderRadius,
+				borderColor,
+				backgroundColor,
+			},
+			setAttributes,
+		} = props;
+
+		// eslint-disable-next-line react-hooks/rules-of-hooks
+		const { enabledCurrencies } = useEnabledCurrencies();
+		// eslint-disable-next-line react-hooks/rules-of-hooks
+		const { isLoading } = useCurrencies();
```

Comment:
Moving this before `useEnabledCurrencies()` might be a good idea, because this is the one fetching the data from the API, that one gets the fetched data which is filled by `useCurrencies()` call.

Reply:
Here https://github.com/Automattic/woocommerce-payments/pull/2883/commits/5bb577253329d5129da6f4afef7231d135133516